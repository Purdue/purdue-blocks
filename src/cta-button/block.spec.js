jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    insertBlock,
} from '@wordpress/e2e-test-utils';

import {
    clickElementByText,
    blockStartup,
    clickCheckbox
} from '../test-helpers'

const block = {blockTitle: 'CTA Button', blockName: 'purdue-blocks/cta-button'}

describe( 'ðŸ”¬ Card Block', () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'ðŸ”Ž Block should be available.', async () => {
        await insertBlock( block.blockTitle )

        // tests that the block is properly inserted and matches the existing snapshot
        expect(await page.$(`[data-type="${block.blockName}"]`)).not.toBeNull()
        expect( await getEditedPostContent()).toMatchSnapshot()
    } )

    describe('ðŸ”¬ Block Editor Fields', () => {
        test( 'ðŸ”Ž Open Media Library Button.', async () => {
            await blockStartup(block)

            // open media library
            await clickElementByText('button', 'Open Media Library')

            // tests that media library does open
            expect(await page.$(`div.media-frame-title`)).not.toBeNull()
        })

        test('ðŸ”Ž Card Text Field', async () => {
            await blockStartup(block)
            
            const typeString = "Test Title"

            await page.focus('input[placeholder="Card text..."]')
            await page.keyboard.type(typeString, {delay: 10})

            const editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"subText":"${typeString}"`)).toBe(true)
        })

        describe('ðŸ”¬ CTA Button Setting Fields', () => {

            test('ðŸ”Ž CTA Button Text Field', async () => {
                await blockStartup(block)
                
                const typeString = "CTA text test"
    
                await page.focus('input[placeholder="CTA Text..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( await getEditedPostContent() ).toMatchSnapshot()
                expect( editedContent.includes(`"ctaText":"${typeString}"`)).toBe(true)
            })

            test('ðŸ”Ž CTA Link Field', async () => {
                await blockStartup(block)
                
                const typeString = "Https://www.purdue.edu"
    
                await page.focus('input[placeholder="Paste permalink or url..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( await getEditedPostContent() ).toMatchSnapshot()
                expect( editedContent.includes(`"link":"${typeString}"`)).toBe(true)
            })
        })
    })

    describe('ðŸ”¬ Side Panel Settings', () => {
        test( 'ðŸ”Ž Image alt text should be editable.', async () => {
            await blockStartup(block)

            const typeString = "Image alt text."
            
            // focus the text box then type into it with the virtual keyboard
            await clickElementByText('label', 'Image Alt Text')
            await page.keyboard.type(typeString, {delay: 10})

            const editedContent = await getEditedPostContent()

            expect( editedContent).toMatchSnapshot()
            expect( editedContent.includes(`"altText":"${typeString}"`)).toBe(true)
        })

        test( 'ðŸ”Ž Open Link in New Tab Chackbox', async () => {
            await blockStartup(block)

            await clickCheckbox('Open link in new tab?')
            
            const editedContent = await getEditedPostContent()

            expect( editedContent).toMatchSnapshot()
            expect( editedContent.includes(`"external":true`)).toBe(true)
        })
    })
})