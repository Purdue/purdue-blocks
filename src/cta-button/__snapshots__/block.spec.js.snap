// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🔬 Card Block 🔎 Block should be available. 1`] = `
"<!-- wp:purdue-blocks/cta-button -->
<div target=\\"_self\\" class=\\"card cta-card\\" rel=\\"noopener noreferrer\\"><div class=\\"card-image\\"><figure class=\\"image\\"><img src=\\"\\" alt=\\"\\"/></figure></div><div class=\\"card-content\\"><p></p><a href=\\"\\" class=\\"cta-card__button\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-button -->"
`;

exports[`🔬 Card Block 🔬 Block Editor Fields 🔎 Card Text Field 1`] = `
"<!-- wp:purdue-blocks/cta-button {\\"subText\\":\\"Test Title\\"} -->
<div target=\\"_self\\" class=\\"card cta-card\\" rel=\\"noopener noreferrer\\"><div class=\\"card-image\\"><figure class=\\"image\\"><img src=\\"\\" alt=\\"\\"/></figure></div><div class=\\"card-content\\"><p>Test Title</p><a href=\\"\\" class=\\"cta-card__button\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-button -->"
`;

exports[`🔬 Card Block 🔬 Block Editor Fields 🔬 CTA Button Setting Fields 🔎 CTA Button Text Field 1`] = `
"<!-- wp:purdue-blocks/cta-button {\\"ctaText\\":\\"CTA text test\\"} -->
<div target=\\"_self\\" class=\\"card cta-card\\" rel=\\"noopener noreferrer\\"><div class=\\"card-image\\"><figure class=\\"image\\"><img src=\\"\\" alt=\\"\\"/></figure></div><div class=\\"card-content\\"><p></p><a href=\\"\\" class=\\"cta-card__button\\">CTA text test</a></div></div>
<!-- /wp:purdue-blocks/cta-button -->"
`;

exports[`🔬 Card Block 🔬 Block Editor Fields 🔬 CTA Button Setting Fields 🔎 CTA Link Field 1`] = `
"<!-- wp:purdue-blocks/cta-button {\\"link\\":\\"Https://www.purdue.edu\\"} -->
<div target=\\"_self\\" class=\\"card cta-card\\" rel=\\"noopener noreferrer\\"><div class=\\"card-image\\"><figure class=\\"image\\"><img src=\\"\\" alt=\\"\\"/></figure></div><div class=\\"card-content\\"><p></p><a href=\\"Https://www.purdue.edu\\" class=\\"cta-card__button\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-button -->"
`;

exports[`🔬 Card Block 🔬 Side Panel Settings 🔎 Image alt text should be editable. 1`] = `
"<!-- wp:purdue-blocks/cta-button {\\"altText\\":\\"Image alt text.\\"} -->
<div target=\\"_self\\" class=\\"card cta-card\\" rel=\\"noopener noreferrer\\"><div class=\\"card-image\\"><figure class=\\"image\\"><img src=\\"\\" alt=\\"Image alt text.\\"/></figure></div><div class=\\"card-content\\"><p></p><a href=\\"\\" class=\\"cta-card__button\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-button -->"
`;

exports[`🔬 Card Block 🔬 Side Panel Settings 🔎 Open Link in New Tab Chackbox 1`] = `
"<!-- wp:purdue-blocks/cta-button {\\"external\\":true} -->
<div target=\\"_blank\\" class=\\"card cta-card\\" rel=\\"noopener noreferrer\\"><div class=\\"card-image\\"><figure class=\\"image\\"><img src=\\"\\" alt=\\"\\"/></figure></div><div class=\\"card-content\\"><p></p><a href=\\"\\" class=\\"cta-card__button\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-button -->"
`;
